envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -18,7 +18,15 @@ option (gogoproto.equal_all) = true;
 
 // Upstream host identifier.
 message Endpoint {
+  // The upstream host address.
   core.Address address = 1;
+
+  // [#not-implemented-hide:] Optional alternative health check port.
+  //
+  // By default the health check address port of an upstream host is the same as the host's serving
+  // address port. This provides an alternative health check port. If set it allows an upstream host
+  // to have different health check address port. The health checker should honor this.
+  uint32 health_check_port = 2;
```

Comment:
I think we might eventually want something closer to https://github.com/envoyproxy/data-plane-api/blob/f88047c0d470352ad1bf6a97dd64155bc65047a0/envoy/api/v2/core/address.proto#L40, but for now this is OK (we can always upgrade to a `oneof` without breaking wire compat).